dist/htmlminifier.js

```diff
@@ -1508,8 +1531,8 @@
           }
           optionalStartTag = '';
           // </html> or </body> may be omitted if not followed by comment
-          // </head> or </colgroup> may be omitted if not followed by space or comment
-          if (optionalEndTag === 'html' || optionalEndTag === 'body' || (optionalEndTag === 'head' || optionalEndTag === 'colgroup') && !/^\s/.test(text)) {
+          // </head>, </colgroup> or </caption> may be omitted if not followed by space or comment
+          if (optionalEndTag === 'html' || optionalEndTag === 'body' || (optionalEndTag === 'head' || optionalEndTag === 'colgroup' || optionalEndTag === 'caption') && !/^\s/.test(text)) {
```

Comment:
If you mean the parens around `head`, `colgroup` and `caption`, then yes, due to precedence of `&&` over `||`, i.e. `!/^\s/.test(text)` applies to all of those inside the parens, but not `head` or `body`.

But I guess another `createMapFromString` is in store here... :stuck_out_tongue_winking_eye: